src/main/java/fi/aalto/cs/apluscourses/intellij/model/SettingsImporterImpl.java

```diff
@@ -91,11 +104,54 @@ public void importProjectSettings(@NotNull Project project, @NotNull Course cour
 
     for (String fileName : fileNames) {
       Path path = Paths.get(fileName);
+      // The ZIP contains a .idea directory with all of the settings files. We want to extract the
+      // files to the .idea directory without the .idea prefix, as otherwise we would end up with
+      // .idea/.idea/<settings_files>.
       Path pathWithoutRoot = path.subpath(1, path.getNameCount());
-      zipFile.extractFile(path.toString(), settingsPath.toString(), pathWithoutRoot.toString());
+      zipFile.extractFile(path.toString(), target.toString(), pathWithoutRoot.toString());
     }
+  }
+
+  private static final String WORKSPACE_XML_COMPONENT_NAME = "CompilerWorkspaceConfiguration";
+  private static final String WORKSPACE_XML_OPTION_NAME = "AUTO_SHOW_ERRORS_IN_EDITOR";
+  private static final String WORKSPACE_XML_OPTION_VALUE = "false";
 
-    ProjectManager.getInstance().reloadProject(project);
+  /**
+   * Parses the XML file (usually the workspace.xml file stored in .idea) at the given path and
+   * returns a {@link Document} with file contents and an additional setting added.
+   * @param workspaceXmlPath The path pointing to the XML file.
+   */
```

Comment:
could you please write a brief note about why and what is being `thrown` "in case of :fire: "?

Reply:
Good idea, will do :+1: